source/main.cc

```diff
@@ -408,6 +408,20 @@ void print_version_information(Stream &stream)
 }
 
 
+// hook into SIGABRT and kill off the program
+void signal_handler(int signal)
+{
+  if (signal == SIGABRT)
+    {
+      std::cerr << "SIGABRT received\n";
+    }
+  else
+    {
+      std::cerr << "Unexpected signal " << signal << " received\n";
+    }
+  std::_Exit(EXIT_FAILURE);
```

Comment:
I can't find this function in C++03 (though it's in C++14). Would we need to call this function conditionally on the compiler being new enough?